src/main/java/de/sakpaas/backend/service/SearchService.java

```diff
@@ -74,28 +78,48 @@ public SearchResultObject search(String query, CoordinateDetails coordinateDetai
    * @param coordinateDetails The SearchCoordinates
    * @return A new SearchRequest
    */
+
+
   protected SearchRequest createRequest(String query, CoordinateDetails coordinateDetails)
       throws EmptySearchQueryException {
-    return new SearchRequest();
-  }
+    String lowerquery = query.toLowerCase();
 
-  /**
-   * Possible Brand Names will be extracted from the Query and saved brands List.
-   *
-   * @param request The Request Object
-   * @return the updated Request Object
-   * @throws EmptySearchQueryException Will be thrown if the Query is
-   *                                   Empty(needs to be implemented)
-   */
-  protected SearchRequest checkForBrands(SearchRequest request) {
-    request.setBrands(
-        request.getQuery().stream().filter(queryElement -> knownBrands.contains(queryElement))
-            .collect(
-                Collectors.toSet()));
-    request.getQuery().removeAll(request.getBrands());
-    return request;
-  }
+    SearchRequest searchRequest = new SearchRequest();
+
+
+    ArrayList<Integer> containsbrand = new ArrayList<Integer>();
 
+    HashSet<String> giveQuery = new HashSet<String>();
+    HashSet<String> giveBrands = new HashSet<String>();
+
+
+    for (String brand : knownBrands) {
+      Pattern word = Pattern.compile(brand);
+      Matcher match = word.matcher(lowerquery);
+
+      if (match.find()) {
```

Comment:
Warum benutzt du hier nicht `string.contains()`? Das ist in diesem Fall besser als die `Pattern.compile()` Lösung, da sonst die `brand` als `regex`-String interpretiert wird und wenn das nicht korrekt geht, dann crasht die ganze Suche oder Verhält sich ganz anders als gedacht, da die Brandnamen in der Datenbank auch Sonderzeichen wie `,-.;/` etc. beinhalten und die würden alle irgendwie interpretiert werden oder halt crashen wenn das nicht geht.